scripts/findUnused.ts

```diff
@@ -1,32 +1,140 @@
 import { relative } from "node:path"
-import { BindingNamedNode, Project } from "ts-morph"
+import { fromHere } from "@re-/node"
+import { BindingNamedNode, Node, Project, ts } from "ts-morph"
 
 const project = new Project({
-    tsConfigFilePath: "tsconfig.references.json"
+    tsConfigFilePath: fromHere("..", "tsconfig.references.json")
 })
 
 const unused: Record<string, string[]> = {}
+const snippetFileRegex = /snippets\/\w+/
+const exportAllRegex = /export \*/
+const exportAllRenamedRegex = /\* as /
+const ignoreUnusedComment = "@ignore-unused"
+const apiExports: ApiExports[] = []
+const sourceFiles = project.getSourceFiles()
+type DeclarationInfo = {
+    filePath: string
+    sourceFileText: string
+    declarationNode: Node<ts.Node>
+    referencesLength: number
+    declarationName: string
+}
+type ApiExports = {
+    path: string
+    exportedDeclarations: string[]
+}
+const checkForIgnoreUnusedComment = ({
+    filePath,
+    sourceFileText,
+    declarationNode,
+    referencesLength,
+    declarationName
+}: DeclarationInfo) => {
+    for (const range of declarationNode.getLeadingCommentRanges()) {
+        if (
+            sourceFileText
+                .slice(range.getPos(), range.getEnd())
+                .includes(ignoreUnusedComment)
+        ) {
+            if (referencesLength > 1) {
+                console.log(
+                    `@ignore-unused wasn't needed for ${declarationName} - ${filePath}\n`
+                )
+            }
+            return true
+        }
+    }
+    return false
+}
 
+const indexFiles = sourceFiles.filter(
+    (file) => file.getBaseName() === "index.ts"
+)
+for (const file of indexFiles) {
+    const publicApiExports = file
+        .getExportDeclarations()
+        .filter((declaration) => !exportAllRegex.test(declaration.getText()))
+        .map((declaration) => {
+            return {
+                path: relative(".", declaration.getSourceFile().getFilePath()),
+                exportedDeclarations: declaration
+                    .getNamedExports()
+                    .map((namedExport) => namedExport.getName())
+            }
+        })
+    if (publicApiExports.length) {
+        apiExports.push(...publicApiExports)
+    }
+}
 for (const sourceFile of project.getSourceFiles()) {
```

Comment:
Let's break the top-level of the script up into a couple descriptively named functions so it's more immediately clear what the major parts of the script do.